backend/custom_models.py

```
@@ -0,0 +1,231 @@
+def _load_config() -> dict:
+    """Load custom models config from disk.
+
+    On IOError the file is simply missing — return an empty config.
+    On JSONDecodeError the file is corrupt — back it up, log the error,
+    and re-raise so callers do not accidentally overwrite it.
+    """
+    path = _get_config_path()
+    if not path.exists():
+        return {"models": []}
+    try:
+        with open(path, "r") as f:
```

Ruff (0.15.10)

[warning] Unnecessary mode argument

Remove mode argument

(UP015)